### ISSUE TITLE
使用Qwen3-0.6B进行embedding微调失败

### ISSUE THREAD

**Issue by jiangh0** [modelscope/ms-swift#5298]
**Describe the bug**
1. 用以下配置进行embedding微调时，Qwen3-Embedding-0.6B训练后评测指标正常，Qwen3-0.6B评测指标异常(ndcg@10仅0.00062)

$torch_distributed_launch swift/cli/sft.py \
    --model $MODEL \
    --model_type qwen3 \
    --torch_dtype bfloat16 \
    --attn_impl flash_attention_2 \
    --task_type embedding \
    --train_type full \
    --dataset $DATASET \
    --split_dataset_ratio 0.05 \
    --eval_strategy steps \
    --output_dir $OUTPUT_DIR \
    --save_steps 200 \
    --eval_steps 100 \
    --num_train_epochs 5 \
    --per_device_train_batch_size 8 \
    --per_device_eval_batch_size 8 \
    --gradient_accumulation_steps 1 \
    --learning_rate 2e-5 \
    --loss_type infonce \
    --max_length 2048 \
    --truncation_strategy right \
    --label_names labels \
    --dataloader_drop_last true

2. 框架支持用moe模型进行embedding和reranker训练吗

**Comment by z0o0ey** [comment]
please retry with latest swift and vllm, feel free to reopen or create a new issue if it the problem persists.
